packages/compiler/src/experimental/typekit/create-diagnosable.ts

```diff
@@ -38,8 +54,8 @@ export interface Diagnosable<P extends unknown[], R> {
  * @experimental
  */
 export function createDiagnosable<P extends unknown[], R>(
-  fn: DiagnosableFunction<P, R>,
-): Diagnosable<P, R> {
+  fn: ThisDiagnosableFunction<P, R>,
```

Comment:
```suggestion
  fn: (this: TypeKit, ...args: P) => [R, readonly Diagnostic[]]
```
so you don't need the extra type and as you already doing it inline below

Reply:
Good call - I was experimenting with that type and forgot it's only used in one space.